backends/advanced/src/advanced_omi_backend/controllers/system_controller.py

```
@@ -562,3 +413,126 @@ async def set_memory_provider(provider: str):
+        yaml_text = yaml.safe_dump({"memory": memory_section}, sort_keys=False)
+        return {"config_yaml": yaml_text}
+    except Exception as e:
+        logger.error(f"Failed to load memory configuration: {e}")
+        return JSONResponse(status_code=500, content={"error": f"Failed to load memory configuration: {str(e)}"})
+
+
+def _parse_memory_yaml(config_yaml: str) -> dict:
+    try:
+        doc = yaml.safe_load(config_yaml) or {}
+        if not isinstance(doc, dict):
+            raise ValueError("YAML must define a mapping")
```

Ruff (0.14.8)

Prefer `TypeError` exception for invalid type

(TRY004)